### PR TITLE
Add task info to player info

### LINKED ISSUE #140: missing TaskInfo data in SetTasks and Tasks in API.IInnerPlayerInfo
https://github.com/Impostor/Impostor/blob/bde577db751333cae56b47acc13e5573f462754c/src/Impostor.Server/Net/Inner/Objects/InnerGameData.cs#L178-L182
for all tasks `Id = index` , `Complete = false`

- [ ] maybe you need to use the code like `InnerPlayerInfo.Deserialize` ?
https://github.com/Impostor/Impostor/blob/49284a587cd51edfb9099422560001abf7c8ef1f/src/Impostor.Server/Net/Inner/Objects/InnerPlayerInfo.cs#L56-L62

### API enhancement
- [x] add `Tasks` property into `Impostor.Api.Net.Inner.Objects.IInnerPlayerInfo`

### PR DESCRIPTION
**Pull request by AlexejheroYTB**
### Description

This PR adds task information to IInnerPlayerInfo.

---

closes #140

### CODE REVIEW

File: src/Impostor.Server/Net/Inner/Objects/InnerGameData.cs [hunk 1]
```diff
@@ -172,12 +172,12 @@ private void SetTasks(byte playerId, ReadOnlyMemory<byte> taskTypeIds)
...
 
-            player.Tasks = new List<TaskInfo>(taskTypeIds.Length);
+            player.Tasks = new List<ITaskInfo>(taskTypeIds.Length);
 
             for (var i = 0; i < taskTypeIds.Length; i++)
             {
-                player.Tasks.Add(new TaskInfo());
-                player.Tasks[i].Id = (uint)i;
+                var task = new TaskInfo { Id = (uint)i };
+                player.Tasks.Add(task);
```
Comment (Mr-VIT) [inline src/Impostor.Server/Net/Inner/Objects/InnerGameData.cs:180]: does not close  #140 
still missing TaskInfo data from `taskTypeIds`

Comment (AlexejheroYTB) [reply src/Impostor.Server/Net/Inner/Objects/InnerGameData.cs:180]: What do you mean?

Comment (Mr-VIT) [reply src/Impostor.Server/Net/Inner/Objects/InnerGameData.cs:180]: player.Tasks is empty need deserialize `taskTypeIds`

Comment (AlexejheroYTB) [reply src/Impostor.Server/Net/Inner/Objects/InnerGameData.cs]: Resolved in 2667387

File: src/Impostor.Api/Net/Inner/Objects/ITaskInfo.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Impostor.Api.Net.Inner.Objects
+{
+    public interface ITaskInfo
+    {
+        uint Id { get; }
+        
+        bool Complete { get; }
+
+        void Serialize(IMessageWriter writer);
```
Comment (AeonLucid) [inline src/Impostor.Api/Net/Inner/Objects/ITaskInfo.cs:11]: I don't think `Serialize` and `Deserialize` should be exposed to the plugin API.

Comment (AlexejheroYTB) [reply src/Impostor.Api/Net/Inner/Objects/ITaskInfo.cs]: Resolved in 312fa43bac9a6538bc99c9d087f1f5ab01308031

**Pull request merged**